core/server/worker/src/main/java/alluxio/worker/grpc/DataMessageServerStreamObserver.java

```diff
@@ -51,7 +51,12 @@ public void onNext(DataMessage<T, DataBuffer> value) {
     if (buffer != null) {
       mBufferRepository.offerBuffer(buffer, value.getMessage());
     }
-    mObserver.onNext(value.getMessage());
+    try {
+      mObserver.onNext(value.getMessage());
+    } catch (Throwable e) {
+      mBufferRepository.pollBuffer(value.getMessage());
```

Comment:
Missing release? The buffer contained in the `DataBuffer` can be a Netty ByteBuf that is reference counted and needs explicit releasing. See `alluxio.network.protocol.databuffer.NettyDataBuffer`

```suggestion
      mBufferRepository.pollBuffer(value.getMessage()).release();
```